jwst/regtest/test_nircam_tsgrism.py

```diff
@@ -5,26 +5,30 @@
 from jwst.lib.set_telescope_pointing import add_wcs
 from jwst.stpipe import Step
 
+# Mark all tests in this module
+pytestmark = [pytest.mark.bigdata]
+
 
 @pytest.fixture(scope="module")
-def run_pipelines(rtdata_module):
+def run_pipelines(rtdata_module, resource_tracker):
     """Run stage 2-3 tso pipelines on NIRCAM TSO grism data."""
     rtdata = rtdata_module
 
-    # Run tso-spec2 pipeline on the _rateints file, saving intermediate products
-    rtdata.get_data("nircam/tsgrism/jw01366002001_04103_00001-seg001_nrcalong_rateints.fits")
-    args = ["calwebb_spec2", rtdata.input,
-            "--steps.flat_field.save_results=True",
-            "--steps.extract_2d.save_results=True",
-            "--steps.srctype.save_results=True"
-            ]
-    Step.from_cmdline(args)
-
-    # Get the level3 association json file (though not its members) and run
-    # the tso3 pipeline on all _calints files listed in association
-    rtdata.get_data("nircam/tsgrism/jw01366-o002_20230107t004627_tso3_00001_asn.json")
-    args = ["calwebb_tso3", rtdata.input]
-    Step.from_cmdline(args)
+    with resource_tracker.track():
+        # Run tso-spec2 pipeline on the _rateints file, saving intermediate products
+        rtdata.get_data("nircam/tsgrism/jw01366002001_04103_00001-seg001_nrcalong_rateints.fits")
+        args = ["calwebb_spec2", rtdata.input,
+                "--steps.flat_field.save_results=True",
+                "--steps.extract_2d.save_results=True",
+                "--steps.srctype.save_results=True"
+                ]
+        Step.from_cmdline(args)
+
+        # Get the level3 association json file (though not its members) and run
+        # the tso3 pipeline on all _calints files listed in association
+        rtdata.get_data("nircam/tsgrism/jw01366-o002_20230107t004627_tso3_00001_asn.json")
+        args = ["calwebb_tso3", rtdata.input]
+        Step.from_cmdline(args)
```

Comment:
should this maybe be split by pipeline like most of the other tests?

Reply:
Yes, there are many inconsistent design choices across our regtests. I can make this change, though a more uniform design would probably be worth looking into at some point, wrt. how steps are called, how the tests interact with the rtdata module, how we test optional inputs, and how we use unique (or not) filenames across tests in a file.

Reply:
I agree some more uniformity would be nice.  On the upside, it looked like this was the only one out of the ones you decided to memory-track that ran more than one pipeline at once - and that's why I noticed it